libfabric build failed in an internal version of Visual Studio

I'm a member of Microsoft vcpkg team. In our latest internal test, libfabric installation failed with following error:
`C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\include\yvals_core.h(23,1): fatal error C1189: #error:  STL1003: Unexpected compiler, expected C++ compiler.`

This issue occured due to the new change: https://github.com/microsoft/STL/pull/2148 of STL. This change will be released in VS 2022 17.1 Preview 1, the corresponding toolset is V143. More info please see: https://github.com/microsoft/STL/wiki/Changelog

If you want reproduce this issue, please download the latest yvals_core.h: https://github.com/microsoft/STL/blob/main/stl/inc/yvals_core.h and replace the local version.

I know the current toolsets used in libfabric are v140, v141 and v142, but if you have plans to support v143, you may need to make some changes to avoid the above error.

--- a/prov/netdir/src/netdir_ndinit.c
+++ b/prov/netdir/src/netdir_ndinit.c
@@ -36,7 +36,7 @@
 #include <guiddef.h>
 
 #include <ws2spi.h>
-#include <cassert>
+#include <assert.h>
 #include "ndspi.h"
 
 #include "netdir.h"

